presubmit.sh

```diff
@@ -38,35 +38,36 @@ fi
 
 if [[ ( ${JOB_ARCHITECTURE} == "" && ${JOB_ENABLE_GL} == "1" ) ]]; then
     BUILD_OPENGL_TEST="ON"
-    sudo apt-get update
-    sudo apt-get -y install libglu1-mesa-dev freeglut3-dev mesa-common-dev libglew-dev
 fi
-# Prepare headers
-git clone https://github.com/KhronosGroup/OpenCL-Headers.git
-cd OpenCL-Headers
-ln -s CL OpenCL # For OSX builds
-cd ..
 
 # Get and build loader
 git clone https://github.com/KhronosGroup/OpenCL-ICD-Loader.git
```

Comment:
Is there a reason to fetch the headers as part of the GitHub actions YAML file but the ICD loader as part of this presubmit script?

Reply:
The headers could be moved very easily whereas moving the ICD loader fetch and build steps to the YAML file requires porting the whole cross-compilation logic. It should be done but it was more than I had time for when looking at this. Over time, I think we should move everything to the YAML file.

Reply:
Cool, makes sense, this PR is a step in the right direction then (even if it breaks @EwanC 's workflow!).

If it's helpful I do fetch, build, and use the ICD loader via the YAML file in one of my projects:

https://github.com/bashbaug/opencl-extension-loader/blob/main/.github/workflows/cmake.yml#L83

It did require some special-casing that was not trivial, and I'm not 100% sure I've done it in the best way...

Reply:
Yup, sorry @EwanC :). Although, I agree with you that we should really write a decent README. A basic build is easy enough, it's the cross compilation logic that'd require a bit more work (toolchain files handling as a separate step, etc).